# Walidacja: zestawy cech na nowych danych (Sprint 4+)

## Cel
Spojna, swieza walidacja walk-forward wszystkich zestawow cech na nowych danych: surface_speed,
fatigue, enriched (surface+fatigue), Elo. Per sezon: baseline (cache) + kazdy zestaw na tych samych
meczach, parowanie + McNemar.

In [1]:
import sys
from pathlib import Path
sys.path.insert(0, str(Path("../src").resolve()))

## Uruchomienie (od zera)

In [2]:
import tennis_model_validate_features as m
m.main()


===== ROK 2022 =====


  baseline match=0.6728


    surface    match=0.6782  delta=+0.0055


    fatigue    match=0.6746  delta=+0.0018


    enriched   match=0.6764  delta=+0.0037


    elo        match=0.6965  delta=+0.0238



===== ROK 2023 =====


  baseline match=0.6346


    surface    match=0.6435  delta=+0.0089


    fatigue    match=0.6381  delta=+0.0036


    enriched   match=0.6346  delta=+0.0000


    elo        match=0.6328  delta=-0.0018



===== ROK 2024 =====


  baseline match=0.6186


    surface    match=0.6237  delta=+0.0051


    fatigue    match=0.6220  delta=+0.0034


    enriched   match=0.6254  delta=+0.0068


    elo        match=0.6356  delta=+0.0169



===== ROK 2025 =====


  baseline match=0.6566


    surface    match=0.6566  delta=+0.0000


    fatigue    match=0.6566  delta=+0.0000


    enriched   match=0.6585  delta=+0.0019


    elo        match=0.6377  delta=-0.0189



WALK-FORWARD CECH (nowe dane 2022-2025) vs baseline

--- surface (3 cech) ---
 year  baseline  variant  delta
 2022    0.6728   0.6782 0.0055
 2023    0.6346   0.6435 0.0089
 2024    0.6186   0.6237 0.0051
 2025    0.6566   0.6566 0.0000
  POOLED (2228): baseline=0.6450  surface=0.6499  delta=+0.0049  (dodatnie 3/4 lat)
  McNemar: b=34 c=45 z=1.13 p=0.2606 -> brak istotnosci

--- fatigue (6 cech) ---
 year  baseline  variant  delta
 2022    0.6728   0.6746 0.0018
 2023    0.6346   0.6381 0.0036
 2024    0.6186   0.6220 0.0034
 2025    0.6566   0.6566 0.0000
  POOLED (2228): baseline=0.6450  fatigue=0.6472  delta=+0.0022  (dodatnie 3/4 lat)
  McNemar: b=37 c=42 z=0.45 p=0.6527 -> brak istotnosci

--- enriched (9 cech) ---
 year  baseline  variant  delta
 2022    0.6728   0.6764 0.0037
 2023    0.6346   0.6346 0.0000
 2024    0.6186   0.6254 0.0068
 2025    0.6566   0.6585 0.0019
  POOLED (2228): baseline=0.6450  enriched=0.6481  delta=+0.0031  (dodatnie 3/4 lat)
  McNemar: b=44 c=51 z=

## Wnioski
ZADEN zestaw cech nie jest istotny statystycznie (wszystkie McNemar p > 0.25). Najlepsze sa
directionally dodatnie (+0.4-0.6 p.p.), ale w granicach szumu. To komplet dowodow, ze nic nie
przebija baseline -- sufit ~65% jest odporny na nowe cechy.